File: zig/fizzbuzz.zig
const std = @import("std");

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();

    for (1..101) |i| {
        if (i % 15 == 0) {
            try stdout.print("FizzBuzz\n", .{});
        } else if (i % 3 == 0) {
            try stdout.print("Fizz\n", .{});
        } else if (i % 5 == 0) {
            try stdout.print("Buzz\n", .{});
        } else {
            try stdout.print("{d}\n", .{i});
        }
    }
}

File: pascal/fizzbuzz.pas
program FizzBuzz;

var
  i: Integer;
  result: String;

begin
  for i := 1 to 100 do
  begin
    if (i mod 15 = 0) then
      result := 'FizzBuzz'
    else if (i mod 3 = 0) then
      result := 'Fizz'
    else if (i mod 5 = 0) then
      result := 'Buzz'
    else
      result := IntToStr(i);

    WriteLn(result);
  end;
end.
